header_test.go

```diff
@@ -348,6 +348,47 @@ func TestRequestRawHeaders(t *testing.T) {
 	})
 }
 
+func TestRequestDisableSpecialHeaders(t *testing.T) {
+	t.Parallel()
+
+	kvs := "Host: foobar\r\n" +
+		"User-Agent: ua\r\n" +
+		"non-special:  val\r\n" +
+		"\r\n"
+
+	var h RequestHeader
+	h.DisableSpecialHeader()
+
+	s := "GET / HTTP/1.0\r\n" + kvs
+	br := bufio.NewReader(bytes.NewBufferString(s))
+	if err := h.Read(br); err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	// Host and User-Agent are not parsed
+	if h.Host() != nil {
```

Comment:
I don't understand why `Host()` should return `nil` here? `Host:` is present in the headers?

Reply:
ah this is because for Host and User-Agent they are saved as separate bytes, not present in the headers `h.h`. When we call `Host()` with special headers disabled it tries to read it from `h.h`. I guess technically parsing is currently unaffected, but I agree this behavior is confusing. I'll push another commit that makes sure to parse all special headers  as regular headers